mldsa/polyvec.c

```diff
@@ -39,13 +39,33 @@ void polyvec_matrix_pointwise_montgomery(polyveck *t,
 /**************************************************************/
 
 void polyvecl_uniform_eta(polyvecl *v, const uint8_t seed[MLDSA_CRHBYTES],
-                          uint16_t nonce)
+                          const uint16_t nonce)
 {
   unsigned int i;
-
+  uint16_t n = nonce;
   for (i = 0; i < MLDSA_L; ++i)
+  __loop__(
+    assigns(i, n, object_whole(v))
+    invariant(i <= MLDSA_L)
+    invariant(forall(k1, 0, i,
+      array_abs_bound(v->vec[k1].coeffs, 0, MLDSA_N, MLDSA_ETA + 1))))
   {
-    poly_uniform_eta(&v->vec[i], seed, nonce++);
+    poly t;
+    poly_uniform_eta(&t, seed, n);
+
+#ifdef CBMC
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
+#endif
+    n++;
+#ifdef CBMC
+#pragma CPROVER check pop
+#endif
```

Comment:
The function is called with `nonce=0` and `nonce=L`, so this can never overflow in practice. I'd prefer a stronger pre-condition over silencing the warning here.

Reply:
I've tried a few iterations of stronger pre-conditions, but I get stuck on `smt2` counter-example failure `SMT2 solver returned non-constant value for variable B2865`. When switching to `bitwuzla` the proof runs indefinitely and thus fails to provide a useful error message. I chatted with @rod-chapman on this, and he proposed this as the work around.

Reply:
Even with `requires(nonce==0)`, can we prove that this algorithm will terminate before `nonce==UINT_MAX`? (or whatever is the max value that will lead to overflow on next time...)

Reply:
Got it! Thank you!

Reply:
@jakemas To make this uniform with `polyveck_uniform_eta`, where the nonce is _not_ 0, can you please use `UINT16_MAX - MLDSA_L` as the bound here?

Reply:
done!